rogue macro detected during lowering

Macro that shouldn't be there detected during lowering.

```rust
//@compile-flags: -frust-incomplete-and-experimental-compiler-do-not-use
macro_rules! inner {
    () => {
        $crate::
    }
}

pub fn main() {
    inner!();
}
```
Program output
```
a.rs:8:5: fatal error: rogue macro detected during lowering
    8 |     inner!();
      |     ^~~~~
compilation terminated.
```

--- a/gcc/rust/expand/rust-macro-expand.cc
+++ b/gcc/rust/expand/rust-macro-expand.cc
@@ -327,7 +327,14 @@ MacroExpander::expand_invoc (AST::MacroInvocation &invoc,
   else
     fragment
       = expand_decl_macro (invoc.get_locus (), invoc_data, *rdef, semicolon);
-
+  // fix: if the expansion is failing, we must replace the marco with an empty
+  // error or node
+  // makes sure that it doesn't panic on Rouge macro (it -> Lowering Phase)
+  // added the parsing errors in gcc/testsuite/rust/compile/issue-4213.rs
+  if (fragment.is_error ())
+    {
+      fragment = AST::Fragment::create_empty ();
+    }
   set_expanded_fragment (std::move (fragment));
 }
 

--- a/gcc/testsuite/rust/compile/issue-4213.rs
+++ b/gcc/testsuite/rust/compile/issue-4213.rs
@@ -1,0 +1,34 @@
+// Test for issue #4213 - rogue macro detected during Lowering.
+
+macro_rules! inner {
+    () => {
+        $crate::    // { dg-error "could not parse path expression segment" }
+    };
+}
+
+macro_rules! multi_arm {
+    (a) => { $crate:: };     // { dg-error "could not parse path expression segment" }
+    (b) => { () };
+}
+
+macro_rules! another_macro {
+    () => { $crate:: }      // { dg-error "could not parse path expression segment" }
+}
+macro_rules! generic_macro {
+    ($t:ty) => {
+        $crate::            // { dg-error "could not parse path expression segment" }
+    };
+}
+
+macro_rules! empty_case {
+    () => {};
+}
+
+pub fn main() {
+    let _ = inner!();
+    let _ = multi_arm!(a);
+    let _ = multi_arm!(b);
+    let _ = another_macro!();
+    let _ = generic_macro!(i32);
+    empty_case!();
+}

